### ISSUE TITLE
Datatype conversion of Dart "color" to "string" when in a "list"

### ISSUE THREAD

**Issue by MajorHodl** [FlutterFlow/flutterflow-issues#1201]
### Has your issue been reported?

- [X] I have searched the existing issues and confirm it has not been reported.
- [X] I give permission for members of the FlutterFlow team to access and test my project for the sole purpose of investigating this issue.

### Current Behavior

When I add a dart "color" field to firebase as an individual field in a document it correctly converts the data type to a "string."

However, when a dart "color" is part of a "list" it does not seem to be converting to a "string" for the firebase "array."

That is my best guess. 

Here is the error code from the browser console:

Uncaught (in promise) Error: Invalid argument (dartObject): Could not convert: Instance of 'Color'
    at Object.throw_ [as throw] (dart_sdk.js:5463:11)
    at Object.jsify (utils.dart:95:5)
    at utils.dart:83:36
    at LinkedMap.new.forEach (dart_sdk.js:28663:11)
    at Object.jsify (utils.dart:82:15)
    at MappedListIterable.new.<anonymous> (utils.dart:58:42)
    at MappedListIterable.new.elementAt (dart_sdk.js:24938:25)
    at ListIterator.new.moveNext (dart_sdk.js:24740:55)
    at JsIterator.next (dart_sdk.js:7500:21)
    at List.of (dart_sdk.js:47363:18)
    at MappedListIterable.new.toList (dart_sdk.js:24547:30)
    at Object.jsifyList (utils.dart:58:65)
    at Object.jsify (utils.dart:77:12)
    at utils.dart:83:36
    at LinkedMap.new.forEach (dart_sdk.js:28663:11)
    at Object.jsify (utils.dart:82:15)
    at MappedListIterable.new.<anonymous> (utils.dart:58:42)
    at MappedListIterable.new.elementAt (dart_sdk.js:24938:25)
    at ListIterator.new.moveNext (dart_sdk.js:24740:55)
    at JsIterator.next (dart_sdk.js:7500:21)
    at List.of (dart_sdk.js:47363:18)
    at MappedListIterable.new.toList (dart_sdk.js:24547:30)
    at Object.jsifyList (utils.dart:58:65)
    at Object.jsify (utils.dart:77:12)
    at utils.dart:83:36
    at LinkedMap.new.forEach (dart_sdk.js:28663:11)
    at Object.jsify (utils.dart:82:15)
    at Object.jsify (utils.dart:42:23)
    at firestore.DocumentReference._fromJsObject.set (firestore.dart:369:46)
    at document_reference_web.dart:33:23
    at Object.guardWebExceptions (_flutterfire_internals.dart:112:21)
    at Object.convertWebExceptions (internals.dart:12:20)
    at document_reference_web.DocumentReferenceWeb.new.set (document_reference_web.dart:32:12)
    at cloud_firestore._JsonDocumentReference.new.set (document_reference.dart:168:22)
    at home_page_widget.dart:192:53
    at Generator.next (<anonymous>)
    at runBody (dart_sdk.js:41451:34)
    at Object._async [as async] (dart_sdk.js:41483:7)
    at home_page_widget.dart:191:28
    at flutter_flow_widgets._FFButtonWidgetState.new.<anonymous> (flutter_flow_widgets.dart:105:32)
    at Generator.next (<anonymous>)
    at runBody (dart_sdk.js:41451:34)
    at Object._async [as async] (dart_sdk.js:41483:7)
    at flutter_flow_widgets.dart:99:15
    at ink_well._InkResponseState.new.handleTap (ink_well.dart:1154:21)
    at tap.TapGestureRecognizer.new.invokeCallback (recognizer.dart:275:24)
    at tap.TapGestureRecognizer.new.handleTapUp (tap.dart:654:11)
    at [_checkUp] (tap.dart:311:5)
    at tap.TapGestureRecognizer.new.acceptGesture (tap.dart:281:7)
    at arena.GestureArenaManager.new.sweep (arena.dart:167:12)
    at binding$5.WidgetsFlutterBinding.new.handleEvent (binding.dart:469:20)
    at binding$5.WidgetsFlutterBinding.new.dispatchEvent (binding.dart:445:14)
    at binding$5.WidgetsFlutterBinding.new.dispatchEvent (binding.dart:331:11)
    at [_handlePointerEventImmediately] (binding.dart:400:7)
    at binding$5.WidgetsFlutterBinding.new.handlePointerEvent (binding.dart:363:5)
    at [_flushPointerEventQueue] (binding.dart:320:7)
    at [_handlePointerDataPacket] (binding.dart:293:9)
    at Object.invoke1 (dart_sdk.js:207087:7)
    at _engine.EnginePlatformDispatcher.new.invokeOnPointerDataPacket (dart_sdk.js:177005:15)
    at [_onPointerData] (dart_sdk.js:178303:49)
    at dart_sdk.js:178901:30
    at dart_sdk.js:178841:9
    at loggedHandler (dart_sdk.js:178480:11)
    at Object._checkAndCall (dart_sdk.js:5666:16)
    at Object.dcall (dart_sdk.js:5671:17)
    at ret (dart_sdk.js:63548:21)
![intercom5](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/5ff30704-26cf-430d-8160-61b86abbfcc8)
![intercom4](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/ffc73b47-2817-48a9-aa4c-d19f5397d868)
![intercom3](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/133a61c6-efa4-4d6c-b626-bfd7638ab611)
![intercom2](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/3bba1186-a7ad-4ae2-b8f6-f12525aad115)
![intercom1](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/7e4e67c7-8703-48a6-b219-7eb78bbfb5fc)

### Expected Behavior

Expected behavior is that the dart "list" containing "color" data should be converted to a "string" in the firebase "array" and saved to the database. 
![intercom5](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/6e637b5d-fc34-442d-a897-20751d58138a)
 

### Steps to Reproduce

Here is a simplified example that will reproduce the problem simply:

https://app.flutterflow.io/project/test-dlx4l6

1) click the "add list 1" button to add data to one list
2) click the "add list 2" button to add the list to a second list
3) click "Firebase" to attempt to save the data to the database

It should throw an error in the web console as described. 

### Reproducible from Blank

- [X] The steps to reproduce above start from a blank project.

### Bug Report Code (Required)

ITESz/Hqx89god9D0bnqau1qhQAkH1x/bZJEl9cbVQgjC4OtE7AXO+r+VExBT/WmdFZEE038mjkf0MLLioX9NcIVPRatUoNoz6VUWjvycHuuMIyrEbuJdkQnM8NTJn6hyqe02R57JtFsWCE5w2z/cq3qNleeY8aSfxBlZ7vfcPo=

### Context

It is not allowing me to save necessary data in order to allow users to customize icons in my project

### Visual documentation

![intercom5](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/b7b0b913-1643-443d-99ad-75921cea9255)
![intercom4](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/efbfa124-da6f-4590-a7c2-c2207c29f4a7)
![intercom3](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/06a973fa-4a97-4b7b-b7d3-f4691919c1d9)
![intercom2](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/5307054a-6190-42d9-ad7d-61c2bd52b6fa)
![intercom1](https://github.com/FlutterFlow/flutterflow-issues/assets/115755916/2d131a91-b8e9-4b16-9e4c-f90fe747bc3d)

### Environment

```markdown
- FlutterFlow version: 
- Platform:
- Browser name and version:
- Operating system and version affected:
```

**Comment by MajorHodl** [comment]
I'm not a proper dev lol. I know writing tickets is an artform. Sorry for the duplicate images. Hopefully this helps. I am fairly confident I have identified a bug but I would be glad to find out the issue is simply a matter of user error.

**Comment by MajorHodl** [comment]
I provided a blank example as well steps, screenshots, and error codes to reproduce.

It is very simple, if you try to save colors in mapped arrays to firebase the data is not being converted to a string.

**Comment by MajorHodl** [comment]
I hope it doesn't get closed. I am a paying customer who has found a bug, has gone out of his way to produce a clear example reproduction of the problem, provided steps to reproduce, as well as several screenshots and have been told I have not provided enough information lol.

**Comment by MajorHodl** [comment]
@kevintonb @victoriahuang1 

I have provided steps, examples, and screenshots.

This is very simple. When a color is added to a list of lists and is trying to be saved to a mapped array in firebase... the color object does not seem to be converting to a string.

If we cannot acknowledge the issue and investigate I will be sure to reach back out to Flutter customer support and/or leadership.

I opened this ticket two weeks ago and have only received a single shallow response that shows what I spent a good deal of time crafting for you all was not even read... Do better.
